src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -66,16 +69,45 @@ public ServiceEndpoint(string connectionString, EndpointType type = EndpointType
             Name = name;
         }
 
-        public ServiceEndpoint(ServiceEndpoint endpoint)
+        /// <summary>
+        /// Aad constructor with key
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="endpoint"></param>
+        /// <param name="credential"></param>
+        public ServiceEndpoint(string key, Uri endpoint, TokenCredential credential) : this(endpoint, credential)
```

Comment:
I think one ctor `ServiceEndpoint(Uri endpoint, TokenCredential credential, EndpointType endpointType, string name)` is enough.

Reply:
Currently I prefer not to set the endpoint type in the key.